### ISSUE TITLE
Optimize xlsx reading

### ISSUE THREAD

**Issue by turicas** [turicas/rows#294]
@israelst said `read_only` could optimize the reading time. Needs to test.

**Comment by israelst** [comment]
If you run the script bellow against a spreadsheet with 600k lines, you will probably see the following result.

```python
from openpyxl import load_workbook
from datetime import datetime


def benchmark(kwargs):
    print(kwargs)
    start = datetime.now()
    wb = load_workbook('huge-spreadsheet.xlsx', **kwargs)
    end = datetime.now()
    print('\tLoading time', end - start)

    sheet = wb['Plan1']

    start = datetime.now()
    for i, row in enumerate(sheet.iter_rows()):
        pass
    end = datetime.now()
    print('\tReading time', end - start)
    print('\tVelocity', round(i/(end - start).seconds), 'row/seg')

benchmark({})
benchmark(dict(data_only=True))
benchmark(dict(data_only=True, read_only=True))
```

Result:
```bash
{}
Loading time 0:03:06.818770
Reading time 0:00:14.778435
Velocity 41527 row/seg

{'data_only': True}
Loading time 0:03:16.700701
Reading time 0:00:16.138065
Velocity 36336 row/seg

{'data_only': True, 'read_only': True}
Loading time 0:00:13.189068
Reading time 0:01:15.010641
Velocity 7752 row/seg
```

**Comment by turicas** [comment]
Thanks for your contribution. :)
I've cherry-picked some commits, rebased, changed some code and merged all the changes on a62f2da2afecc42af8ae30664378e8aec7fa1e66.
